libobs/obs.h

```diff
@@ -2518,6 +2518,19 @@ EXPORT void obs_source_frame_copy(struct obs_source_frame *dst,
 /* Get source icon type */
 EXPORT enum obs_icon_type obs_source_get_icon_type(const char *id);
 
+/* ------------------------------------------------------------------------- */
+/* Functions to show notification in main UI */
+enum obs_notify_type {
+	OBS_NOTIFY_TYPE_INFO,
+	OBS_NOTIFY_TYPE_WARNING,
+	OBS_NOTIFY_TYPE_ERROR,
+};
+
+EXPORT uint32_t obs_show_notification(enum obs_notify_type type,
+				      const char *message, bool persist,
```

Comment:
Hm, I feel like this API is missing a few important fields... usually, when it comes to notifications, the following information is passed by API consumers:

 * id → string
 * title → string
 * message → string (alternative names: body, description)
 * priority → enum obs_notification_priority
 * actions → array of ( action name → string )
 * default action → string
 * user data → raw pointer

Actions in notifications are useful to make them actionable. This way, we avoid notifying users about something without giving them a means to act on what is being notified. What do you think?

Reply:
See https://github.com/obsproject/obs-studio/pull/6815#issuecomment-1195702976